niworkflows/utils/spaces.py

```diff
@@ -457,115 +453,128 @@ def __str__(self):
+            self._refs.insert(index, self.check_space(value))
         elif error is True:
             raise ValueError('space "%s" already in spaces.' % str(value))
 
-    def get_std_spaces(self, only_names=True, dim=(2, 3)):
+    def get_spaces(self, standard=True, nonstandard=True, dim=(2, 3)):
         """
-        Return standard spaces.
+        Return space names.
 
         Parameters
         ----------
-        only_names : :obj:`bool`, optional
-            If ``True`` (default), return all unique :class:`Space` names. If ``False``,
-            return all :class:`Space` entities.
+        standard : :obj:`bool`, optional
+            Return standard spaces.
+        nonstandard : :obj:`bool`, optional
+            Return nonstandard spaces.
         dim : :obj:`tuple`, optional
             Desired dimensions of the standard spaces (default is ``(2, 3)``)
 
         Examples
         --------
         >>> spaces = SpatialReferences(['MNI152NLin6Asym', ("fsaverage", {"den": "10k"})])
-        >>> spaces.get_std_spaces()
+        >>> spaces.get_spaces()
         ['MNI152NLin6Asym', 'fsaverage']
 
-        >>> spaces.get_std_spaces(only_names=False)
-        [Space(name='MNI152NLin6Asym', spec={}),
-         Space(name='fsaverage', spec={'den': '10k'})]
+        >>> spaces.get_spaces(standard=False)
+        []
 
-        >>> spaces.get_std_spaces(dim=(3,))
+        >>> spaces.get_spaces(dim=(3,))
         ['MNI152NLin6Asym']
 
         >>> spaces.add(('MNI152NLin6Asym', {'res': '2'}))
-        >>> spaces.get_std_spaces()
+        >>> spaces.get_spaces()
+        ['MNI152NLin6Asym', 'fsaverage']
+
+        >>> spaces.add(('func', {}))
+        >>> spaces.get_spaces()
+        ['MNI152NLin6Asym', 'fsaverage', 'func']
+
+        >>> spaces.get_spaces(nonstandard=False)
         ['MNI152NLin6Asym', 'fsaverage']
 
-        >>> spaces.get_std_spaces(only_names=False)
-        [Space(name='MNI152NLin6Asym', spec={}),
-         Space(name='fsaverage', spec={'den': '10k'}),
-         Space(name='MNI152NLin6Asym', spec={'res': '2'})]
+        >>> spaces.get_spaces(standard=False)
+        ['func']
 
         """
         out = []
-        for s in self.spaces:
-            if s.standard and s.dim in dim:
-                if only_names:
-                    if s.fullname not in out:
-                        out.append(s.fullname)
-                    continue
-                out.append(s)
+        for s in self.references:
+            if (
+                s.fullname not in out
+                and (s.standard is standard or s.standard is not nonstandard)
+                and s.dim in dim
+            ):
+                out.append(s.fullname)
         return out
 
-    def get_templates(self, dim=(2, 3)):
-        """Return output spaces."""
-        return [s for s in self._spaces
-                if s.standard and hasspec(s.spec, ('res', 'den')) and s.dim in dim]
+    def get_standard(self, full_spec=False, dim=(2, 3)):
```

Comment:
we should consider changing this to plural, as it's supposed to be an iterable anyways
```suggestion
    def get_standard(self, full_spec=False, dims=(2, 3)):
```

Reply:
I think this is just a selector... what I want to include is a "tuplefy" method, so that it also accepts ints - e.g., ``dim=3``

Reply:
i think `dims` is more intuitive as it can be more than one.

Reply:
would you change it in the `Reference` object as well?